### DISCUSSION TITLE
Git storage permission denied problem - mkdir '/Users'

### DISCUSSION THREAD

**Discussion by adenoz** [requarks/wiki#3240] [Help / Questions]
I installed wiki.js in DigitalOcean via the Droplet and it works great. However when attempting to setup a storage target via git to Github I run into this problem after applying changes which initially at the top of the screen everything is green but the Status at the bottom of the middle column remains red and has this error message.

- EACCES: Permission denied, mkdir '/Users'. 

I have made all the subsequent children folders I use have read/write permissions for everyone. I am unable to make the actual '/Users' parent directory itself read/write (even for my administrator account) which surely wiki.js does not need. The issues must be somewhere else right.

I am attempting to use Github via ssh and I am confident the ssh configuration is fine. I am not sure if it is relevant but the Github repo is a private one. I followed the instructions [here](https://docs.requarks.io/storage/git) exactly as far as I can tell. I even moved my public ssh key out of my .ssh folder into another folder under my home directory to make sure wiki.js can access it.

There is content in wiki.js but nothing in the Github repo.

I created a wikijs folder as the assigned local repo folder target as well and that folder has read/write access. It appears that wiki.js is having issues writing to my local disk but I cannot work out why. The identified folders in the config not only are full read/write but all enclosed item have the same permissions. I can only assume there is something I am missing.

**Expected behavior**
I am expecting a successful connection to Github for storage.

**Screenshots**
<img width="822" alt="Screen Shot 2021-03-16 at 11 02 28 pm" src="https://user-images.githubusercontent.com/80598992/111367655-63b26800-86ae-11eb-8317-76136d97fa4e.png">
<img width="815" alt="Screen Shot 2021-03-16 at 11 02 43 pm" src="https://user-images.githubusercontent.com/80598992/111367683-6a40df80-86ae-11eb-86e7-0ad19b619f7d.png">

**Host Info (please complete the following information):**
 - OS: Local device is MacOS 10.15.7 and DigitalOcean droplet is Wiki.js 2.4.107 on Ubuntu 18.04.
 - Wiki.js version: Wiki.js 2.5.191 and Node.js 14.16.0
 - Database engine: PostgreSQL 11.8

**Additional context**
I have been amending the folder permissions via the Finder GUI.

**Comment by NGPixel** [comment]
The key can be inserted directly via the UI by changing the "SSH Private Key Mode" to "Contents"

As for the repo path permissions issue, I would suggest using a path outside the /users directory (e.g. /wiki-data) and set this directory writable to the wiki.js process.

**Comment by adenoz** [comment]
Firstly, many thanks for the advice. I am pretty sure the ssh is fine as I made the ssh config change and it was still throwing the permission denied mkdir error for whatever high level root folder I was attempting to try. I tried manually creating a new folder at the root level (to the side of the /Users/ directory and none of them allowed me to set read-write permissions or the system wouldn't let me even create one. After searching around on the internet I came across this interesting announcement from Apple about improvements from Catalina.

[About the read-only system volume in macOS Catalina](https://support.apple.com/en-au/HT210650)

What appears to have happened is that Apple made basically the entire volume read-only. This seems to be the underlying issue regarding my endless permissions error mkdir errors no matter where I was trying to save the wiki.js git repo.

I've tried setting up another directory at the root level but have so far been unable. I am not an expert in this at all so I may be missing something simple or obvious. I'll keep trying for now as I really want to be able to use wiki.js with git and github.

I'm happy to hear any other suggestions. I'll keep trying to find read-write directory solutions as my issue doesn't seem to be related to wiki.js from what I can tell. I'm happy to leave this issue open for now to take on suggestions from others and if I find a solution to this new and improved highly secure macos read-only volume I'll be sure to update here. I can't help feeling I'm missing something simple as I haven't seen this mentioned anywhere else and I feel like it must have effected other users if my apparent predicament was a real unsolvable thing.
